src/extensions/askUserForParam.ts

```diff
@@ -47,28 +47,26 @@ export function checkResponse(response: string, spec: Param): boolean {
   if (spec.validationRegex && !!response) {
     // !!response to ignore empty optional params
     const re = new RegExp(spec.validationRegex);
-    _.forEach(responses, (resp) => {
+    for (const resp of responses) {
       if ((spec.required || resp !== "") && !re.test(resp)) {
         const genericWarn =
           `${resp} is not a valid value for ${spec.param} since it` +
           ` does not meet the requirements of the regex validation: "${spec.validationRegex}"`;
         utils.logWarning(spec.validationErrorMessage || genericWarn);
         valid = false;
       }
-    });
+    }
   }
 
   if (spec.type && (spec.type === ParamType.MULTISELECT || spec.type === ParamType.SELECT)) {
-    _.forEach(responses, (r) => {
+    for (const r of responses) {
```

Comment:
Is it reliable to assume that if a response was not provided, then it wasn't required? Maybe we want to check the spec.params to verify required/not required - for example, in case the frontend had a bug.